Core/src/main/java/com/paypal/android/core/analytics/DeviceInspector.kt

```diff
@@ -0,0 +1,63 @@
+package com.paypal.android.core.analytics
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import com.paypal.android.core.BuildConfig
+import com.paypal.android.core.analytics.models.DeviceData
+
+class DeviceInspector internal constructor(
+    private val clientSDKVersion: String,
+    private val sdkInt: Int,
+    private val deviceManufacturer: String,
+    private val deviceModel: String,
+    private val deviceProduct: String,
+    private val deviceFingerprint: String
+) {
+
+    constructor() : this(
+        clientSDKVersion = BuildConfig.PAYPAL_SDK_VERSION,
+        sdkInt = Build.VERSION.SDK_INT,
+        deviceManufacturer = Build.MANUFACTURER,
+        deviceModel = Build.MODEL,
+        deviceProduct = Build.PRODUCT,
+        deviceFingerprint = Build.FINGERPRINT
+    )
+
+    fun inspect(context: Context): DeviceData {
+        val packageManager = context.packageManager
+
+        val appName = try {
+            val applicationInfo = packageManager.getApplicationInfo(context.packageName, 0)
+            packageManager.getApplicationLabel(applicationInfo).toString()
+        } catch (ignored: PackageManager.NameNotFoundException) {
+            ""
+        }
+
+        val merchantAppVersion = try {
+            val packageInfo = packageManager.getPackageInfo(context.packageName, 0)
+            packageInfo.versionName
+        } catch (ignored: PackageManager.NameNotFoundException) {
+            ""
+        }
+
+        val appId = context.packageName
+        val clientOS = "Android API $sdkInt"
+
+        val isSimulator = "google_sdk".equals(deviceProduct, ignoreCase = true) ||
+                "sdk".equals(deviceProduct, ignoreCase = true) ||
+                "Genymotion".equals(deviceManufacturer, ignoreCase = true) ||
+                deviceFingerprint.contains("generic")
+
+        return DeviceData(
```

Comment:
Instead of needing an additional type `DeviceData`, can we instead have getters on `DeviceInspector` for these properties?

Just thinking if there's a way we could simplify a bit. I think `DeviceData` vs `DeviceInspector` is a bit confusing.

Reply:
@jcnoriega brought up the same concern [here](https://github.com/paypal/Android-SDK/pull/104/files/4185eff824ffaabe45b1eaad76afe127f8510dc5#r1026630726). Overall I agree, except that it has potential to make testing more difficult.

To make it simpler, I would remove the `DeviceInspector` class and add a `DeviceData` constructor that takes a `Context`.

In the current design, something about having properties on the device inspector feels odd since it only has one job to "inspect" a given context. I'd be down to pair with y'all to get it merged in quicker and align our design principles.

Reply:
@scannillo @jcnoriega ended up removing `DeviceData`.

Reply:
👋 Sorry for the back and forth. I'm realizing now my suggestion to consolidate is an anti-pattern for the SDK and we prefer the `data class` types not to have logic and really only just do formatting if necessary (like `AnalyticsEventData`).

Reply:
@kgangineni - what do you think? Revert back to `DeviceData`?

Reply:
moved context out from `API` to `DeviceInspector` returning `DeviceData`